forte/data/base_pack.py

```diff
@@ -806,8 +806,10 @@ def is_created_by(
         if isinstance(components, str):
             components = [components]
 
+        entry_tid = entry.tid if isinstance(entry, Entry) else entry
```

Comment:
maybe it is easier/faster if we don't check `entry` or `int` here, but create different functions so the caller can decide which to call.

Reply:
So with regards to Forte, the argument entry can be converted to just `int` which represents `tid`. We only keep it `Union[Entry, int]` to ensure backward compatibility.

Also, I feel that making two separate functions would introduce some redundancy. Should I still make two separate functions?

Reply:
https://github.com/asyml/forte/issues/916